core/src/telemetry/mod.rs

```diff
@@ -3,19 +3,21 @@
 
 mod log_export;
 pub(crate) mod metrics;
+#[cfg(feature = "telemetry")]
 mod prometheus_server;
 
-pub use metrics::{
-    build_otlp_metric_exporter, default_buckets_for, start_prometheus_metric_exporter,
-    MetricsCallBuffer,
-};
+#[cfg(feature = "telemetry")]
```

Comment:
Same here about reducing the number of guards. Probably the same module can be used from both places. Likely an `otel` submodule.